lockbox-ios/Store/AutoLockStore.swift

```diff
@@ -114,7 +117,9 @@ class AutoLockStore {
     }
 
     @objc private func lockApp() {
-        self.dataStoreActionHandler.invoke(.lock)
-        self.userDefaults.removeObject(forKey: SettingKey.autoLockTimerDate.rawValue)
+        DispatchQueue.main.async {
+            self.dataStoreActionHandler.invoke(.lock)
```

Comment:
I can't think of a reason that the userdefaults setting or lock function would need to happen on the main thread, can you justify this? I believe you about the timer :)

Reply:
@sashei hmmm not sure. I'll double check this later. My thought was that it would need to eventually get to the main thread to update the UI to show the lock screen.

Reply:
I think if there's not a reason for main-thread dispatching here, I'd rather have the listeners bind to the UI thread when they _receive_ events.

Reply:
looks like this was unnecessary anyways - thanks for calling that out. I've removed it from the PR.